src/opengl/adapters.cpp

```diff
@@ -470,13 +477,15 @@ class GlslFragmentShaderImplementation : public GlslFragmentShader {
         glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, vertexIndexBuffer);
         CHECK_ERROR(__FUNCTION__);
 
+        glBindVertexArray(vao);
         glEnableVertexAttribArray(aVertexData);
         glVertexAttribPointer(aVertexData, 4, GL_FLOAT, GL_FALSE, 0, nullptr);
         CHECK_ERROR(__FUNCTION__);
     }
 
     void unbind() final {
         glDisableVertexAttribArray(aVertexData);
+        glBindVertexArray(0);
```

Comment:
Hmm. Did you try moving this to the end of `unbind` and `glBindVertexArray` to the beginning of `bind` (before the first `glBindBuffer`)? It's unclear to me, if the VAO is needed for `glBindBuffer` or `glEnableVertexAttribArray`... should probably RTFM (i.e., the OpenGL spec)